### PR TITLE
add test for bug 2314

### PR DESCRIPTION
**Pull request by graydon**
Followup to #2316 -- adds a test that fails before and passes after, also a couple new testing interfaces.

### CODE REVIEW

**Comment by MonsieurNicolas** [comment]
@graydon any update on this one?

File: src/bucket/test/BucketManagerTests.cpp [hunk 1]
```diff
@@ -510,6 +510,72 @@ TEST_CASE("bucketmanager reattach to running merge", "[bucket][bucketmanager]")
...
+        bl.resolveAnyReadyFutures();
+    }
+    bm.forgetUnreferencedBuckets();
+    auto bmRefBuckets = bm.getReferencedBuckets();
+    auto bmDirBuckets = bm.getBucketHashesInBucketDirForTesting();
+
+    // Remove the 0 bucket in case it's "referenced"; it's never a file.
+    bmRefBuckets.erase(Hash());
+
+    for (auto const& bmr : bmRefBuckets)
```
[the remark this answers is not in the document]
Comment (graydon) [reply src/bucket/test/BucketManagerTests.cpp]: Fixed.

File: src/bucket/test/BucketManagerTests.cpp [hunk 2]
```diff
@@ -510,6 +510,72 @@ TEST_CASE("bucketmanager reattach to running merge", "[bucket][bucketmanager]")
...
+        }
+        lm.setNextLedgerEntryBatchForBucketTesting({}, {}, dead);
+        closeLedger(*app);
+    }
+
+    // Now check that the bucket dir has only as many buckets as
+    // we expect given the nonzero entries in the bucketlist.
+    while (!bl.futuresAllResolved())
+    {
+        bl.resolveAnyReadyFutures();
```
[the remark this answers is not in the document]
Comment (graydon) [reply src/bucket/test/BucketManagerTests.cpp]: fixed.

**Comment by MonsieurNicolas** [comment]
r+ 19a61f9e99f5fc91afe19a91c50cf953b034bbea

**Pull request merged**